### PR TITLE
add alias descriptors for codechecker

### PR DESCRIPTION
**Pull request by ferkulat**
CodeChecker can run clang-tidy and clang static analysis in one run and
put all warnings in the same file.
Or the user can choose to run either clang-tidy or clang static analysis.

Both are different analysis types with different scopes and
likelihood of false positives.
They should be displayed as different results on jenkins.

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

 Since its just an alias I would claim the CodeCheckerParser tests are sufficient.

### CODE REVIEW

File: SUPPORTED-FORMATS.md [hunk 1]
```diff
@@ -445,6 +504,20 @@ If your tool is supported, but some properties are missing (icon, URL, etc.), pl
...
+            <td>
+                -
+            </td>
+            <td>
+                Cpplint
+            </td>
+            <td>
+                -
+            </td>
+        </tr>
```
Comment (uhafner) [inline SUPPORTED-FORMATS.md:520]: You might not have noticed line 1:
```
<!--- DO NOT EDIT - Generated by ParserRegistry at 2021-11-24T08:25:06.342217-->
```
You need to provide the description in the class.

File: SUPPORTED-FORMATS.md [hunk 2]
```diff
@@ -417,6 +403,79 @@ If your tool is supported, but some properties are missing (icon, URL, etc.), pl
...
+
+Both are different analysis types with different scopes and
+likelihood of false positives.
+This name alias enables warnings-ng-plugin to parse the same CodeChecker report.txt as a distinct tool.
+<p>
+<pre><code>
+steps{
+    sh 'CodeChecker analyze compilecommands.json --analyzers clang-tidy --output dir_to_write_plist_files_into'
+    sh 'CodeChecker parse dir_to_write_plist_files_into > report.txt'
+    recordIssues tool:analysisParser(id: 'code-checker-clang-tidy', pattern: 'report.txt')
```
Comment (uhafner) [inline SUPPORTED-FORMATS.md:475]: The documentation must not use Jenkins internals since this library is used by Github Actions as well.

**Comment by uhafner** [comment]
Actually I do not understand the reasoning for this PR. What exactly are you trying to achieve? Creating an alias without a differentiation does not make much sense?

**Comment by ferkulat** [comment]
CodeChecker creates the same output format for both analysis types: clang-tidy and clang static analysis.
Both types are different.
If I run code checker two times, creating `static_analysis_report.txt` and `clang-tidy_report.txt` and then in the same pipeline would do
~~~~
recordIssues tool:analysisParser(id: 'code-checker', pattern: 'static_analysis_report.txt', name: 'static analysis')
recordIssues tool:analysisParser(id: 'code-checker', pattern: 'clang-tidy_report.txt', name: 'clang tidy')
~~~~
the warnings-ng-plugin would see 'code-checker' in both calls and would try to display both results in the same data set.
Since the report_format is the same for both, I don't need to create a new class for each with the same repeated code.
For the warnings-ng-pluin the different `id` is enough to treat it as different tools and creates a data set for each.

**Comment by ferkulat** [comment]
CodeChecker also supports running both analysis types in one run and put everything into one report.
But that is not what I want. I want them treated as different reports.

**Comment by uhafner** [comment]
But that is much simpler to achieve. There are three options: 

1. use the symbol of your parser:

```
recordIssues tool:codeChecker(id: 'code-checker-analysis', pattern: 'static_analysis_report.txt', name: 'static analysis')
recordIssues tool:codeChecker(id: 'code-checker-tidy', pattern: 'clang-tidy_report.txt', name: 'clang tidy')
```

2. Create new `Tool` classes in the warnings plugin.
3. Wait for (or provide) a bugfix: I accidentally shaded the `id` in the `AnalysisModelParser`. I need to rename that parameter to `registeredId` then you simply can change the ID using the parser in the following way:
```
recordIssues tools: [
      analysisParser(registeredId: 'code-checker', id: 'code-checker-analysis', pattern: 'static_analysis_report.txt', name: 'static analysis')
      analysisParser(registeredId: 'code-checker', id: 'code-checker-tidy', pattern: 'clang-tidy_report.txt', name: 'clang tidy')
      ]
```

**Comment by ferkulat** [comment]
ok Thanks a lot for this hint. That works.

**Pull request closed without merging**